### ISSUE TITLE
v4.4.2 of the eslint plugin doesn't work under node v14

### ISSUE THREAD

**Issue by timbeadle** [makenowjust-labs/recheck#708]
# What is the problem?

- On my projects that use `eslint-plugin-redos`, Dependabot opened upgrade PRs for v4.4.2 of the plugin.
- My projects' CI checks fail under node v14, as it doesn't understand the `||=` operator on L22 of lib/main.js:
```javascript
for (var prop in b ||= {})
```

# How to fix the problem?

1. (For me) Wait until node v14 is end-of-life in April 2023, then upgrade to the new version of the plugin and publish a major version update to my projects.
2. (For you) Unpublish 4.4.2 and publish 5.0.0, as this is a breaking change?
3. Update (or add) the `engines` field in package.json to accurately reflect the node version support.

# Related issues

**Comment by makenowjust** [comment]
@timbeadle Sorry for the late response. I missed this issue.

This issue is completely unexpected. Thus, I will try to fix it.

TODO:

- [ ] Fix for working under Node v14.
  * I guess it will be fixed by modifying the bundler or `tsc` config.
- [ ] Set the correct value to the `engines` field.
  * But, it is hard to maintain.
- [ ] Add tests for some Node versions before releasing as a CI task.

Thanks for reporting.

**Comment by timbeadle** [comment]
@makenowjust Thanks for responding.

For maintaining the `engines` field, take a look at using the `ls-engines` package in your CI tests: this reports whether your engines field matches your dependencies tree's engine support.

https://npmjs.com/package/ls-engines

**Comment by makenowjust** [comment]
Related issue: https://github.com/evanw/esbuild/issues/2628
